### ISSUE TITLE
Buggy JAX rewrite `boolean_indexing_sum`

### ISSUE THREAD

**Issue by ricardoV94** [pymc-devs/pytensor#337]
This rewrite does not check which axis are being summed, and assumes we are always dealing with a complete sum reduction.

https://github.com/pymc-devs/pytensor/blob/8cf3b20ff6e26ee07fe8a0b61fd73e35a6f2d29d/pytensor/tensor/rewriting/jax.py#L49-L80

**Comment by ricardoV94** [comment]
Actually in the valid case, this should be fine, because the advanced boolean index must return a vector. However, there could be mixed indexing going on, in which case the rewrite is again invalid.
